plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/internal/N4JSEclipseModel.java

```diff
@@ -80,15 +80,15 @@ public N4JSEclipseProject getN4JSProject(URI location) {
 		}
 
 		final String projectName = location.lastSegment();
-		final IProject project;
+		IProject project;
+		project = workspace.getProject(projectName);
 		if (location.isFile()) {
 			project = externalLibraryWorkspace.getProject(projectName);
 			checkNotNull(project, "Project does not exist in external workspace. URI: " + location);
```

Comment:
with the refactoring on this branch Source Map View throws exceptions (internally) for every N4JS active editor
```java
java.lang.NullPointerException: Project does not exist in external workspace. URI: file:/Users/me/workspace/myproject
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:226)
	at org.eclipse.n4js.ui.internal.N4JSEclipseModel.getN4JSProject(N4JSEclipseModel.java:87)
	at org.eclipse.n4js.ui.internal.N4JSEclipseModel.getN4JSProject(N4JSEclipseModel.java:1)
	at org.eclipse.n4js.internal.N4JSModel.findProjectWith(N4JSModel.java:125)
	at org.eclipse.n4js.ui.internal.N4JSEclipseModel.findProjectWith(N4JSEclipseModel.java:102)
	at org.eclipse.n4js.ui.internal.N4JSEclipseModel.findN4JSSourceContainer(N4JSEclipseModel.java:130)
	at org.eclipse.n4js.ui.internal.N4JSEclipseCore.findN4JSSourceContainer(N4JSEclipseCore.java:103)
	at org.eclipse.n4js.transpiler.sourcemap.SourceMapFileLocator.resolveSourceMapFromSrc(SourceMapFileLocator.java:39)
	at org.eclipse.n4js.smith.ui.SourceMapView.resolveFromSrc(SourceMapView.java:245)
	at org.eclipse.n4js.smith.ui.SourceMapView.updateActiveEditor(SourceMapView.java:232)
```